Wp-solutions-PS0-2289 - Checking plugin installation and active state

krsomayagi opened the pull request:
## Proposed changes

Code is added  to verify if the plugins are installed and active 

## Type of Change

<!-- What types of changes does your code introduce? -->
<!-- _Put an `x` in the boxes that apply_ -->

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)

## Video

<!-- Add a short video demonstrating where the change takes effect and how to can be reproduced by reviewers -->
<!-- On macOS press cmd+shift+5 to open the screen recording tool. It will save the video to the desktop  -->

<!-- Drag and drop the video file into the GitHub editor to attach it -->

## Checklist

<!-- _Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._ -->

- [ ] I have read the [CONTRIBUTING](https://github.com/bluehost/.github/blob/master/.github/contributing.md) doc
- [ ] I have viewed my change in a web-browser
- [ ] Linting and tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

## Further comments

<!-- If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc... -->

circlecube commented:
@krsomayagi thanks for submitting. I saw this test wasn't passing in the runner. I merged `main` into this branch, and I still don't see this new test passing. Can you take another look? Is it passing for you locally?

krsomayagi commented:
> @krsomayagi thanks for submitting. I saw this test wasn't passing in the runner. I merged `main` into this branch, and I still don't see this new test passing. Can you take another look? Is it passing for you locally?

Sure. Thank you.

circlecube commented:
@krsomayagi—This was outdated, so I updated it from `main` and updated your test file (along with some code tweaks). They are now passing, please take a look and let me know if you have any questions.

krsomayagi commented:
> @krsomayagi—This was outdated, so I updated it from `main` and updated your test file (along with some code tweaks). They are now passing, please take a look and let me know if you have any questions.

@circlecube 
Thank you!

circlecube commented on includes/js/myPluginsTools.js:
```diff
@@ -1,151 +1,177 @@
+		if ( pluginData?.plsProviderName && pluginData?.plsSlug ) {
+			return `<button
+          title="${ isInstalled ? 'Activate' : 'Install' } Premium Plugin"
+          class="nfd-solutions-availble-list-item-button nfd-solutions-${ this.renderNameAsClass(
```
I added this little method `renderNameAsClass` to give us unique class names for each plugin (which helps select them in tests). I also update the order of these installer attributes so they are consistent in each block. The rest of the file changes are due to lint whitespace edits.

circlecube commented on includes/js/myPluginsTools.js:
```diff
@@ -1,151 +1,177 @@
-          href="${this.renderCTAUrl(pluginData?.cta?.url)}"
-          class="nfd-solutions-availble-list-item-button"
-        >${pluginData?.cta?.text}</a>`;
-      } else if ( isInstalled ){ // already installed
```
I removed this isActive but not isInstalled section, and instead use the later conditionals to just check if it is a premium/pls or free plugin. Since those are the more pertinent deciders of what the markup should be. I did file a ticket with the installer module though, since if a plugin is installed and not active, the installer opens up and fails to install since the directory already exists. This change doesn't change the issue, but simplifies the logic in rendering these buttons.

circlecube reviewed (approved):
I'm approving, but would like someone else to take a look since most of this PR ended up being my own code. I've left comments as to the changes in the code. Sorry for all the whitespace/lint cleanup included.

circlecube commented:
Heres a video of the plugins Plugins and Tools page before this update:
https://github.com/user-attachments/assets/9b7b1541-fb30-4a32-ac51-9dbcef43966d

and one with the update:
https://github.com/user-attachments/assets/140cdb34-2373-4b63-b23e-37ddfed3059a

I updated the button when the plugin is already installed to still include the installer attributes for free or pls pluigns, the installer still needs to be updated so it doesn't fail when a plugin is already installed but inactive, ideally it can detect that it is already installed and just activate and redirect. That is in a different module though and beyond scope of this PR. Here we're at least getting a more clear error message that the directory already exists - before it errored and said the API returned a non-success code.
